wpilibj/src/shared/java/edu/wpi/first/wpilibj/command/Scheduler.java

```diff
@@ -41,7 +41,10 @@
    * @return the {@link Scheduler}
    */
   public static synchronized Scheduler getInstance() {
-    return instance == null ? instance = new Scheduler() : instance;
+    if (instance == null) {
```

Comment:
Agree. The right side here is relying on the fact that assignment evaluates to the rhs, and that's giving me flashbacks to my first `if (a = 1)` bugs.